nautobot/extras/tests/test_datasources.py

```diff
@@ -662,3 +685,87 @@ def populate_repo(path, url):
                 device = Device.objects.get(name=self.device.name)
                 self.assertIsNone(device.local_context_data)
                 self.assertIsNone(device.local_context_data_owner)
+
+    def test_git_create_and_dry_run(self, MockGitRepo):
+        with tempfile.TemporaryDirectory() as tempdir:
+            with self.settings(GIT_ROOT=tempdir):
+                MockGitRepo.side_effect = self.populate_repo
```

Comment:
I think I'm missing something here - won't this result in the local directory being populated with files when the MockGitRepo is instantiated?

Reply:
No, only after a `pull_git_repository_and_refresh_data()` is called

Reply:
Yes, but Im testing Database modifications of ConfigContext, ExportTemplates & Device, not if the dirs/files exists.

Reply:
Ive updated the test